File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        output = [0]

        # Go through each number and count the number of bits
        for number in range(1, n + 1):
            num_bits = 0
            temp_num = number
            while temp_num != 0:
                if temp_num & 1 == 1: num_bits += 1
                temp_num = temp_num >> 1
            output.append(num_bits)
        
        return output

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // Create a hashmap to keep track of the count of each number
        HashMap<Integer, Integer> numberCount = new HashMap<>();

        // Go through each number, if it exists, return true, else
        // add it to the hashmap
        for (int number: nums){
            if(numberCount.containsKey(number)){
                return true;
            }
            numberCount.put(number, 1);
        }

        // Return false if we never returned true
        return false;
    }
}
